src/idl_gen_cpp.cpp

```diff
@@ -708,6 +708,23 @@ class CppGenerator : public BaseGenerator {
     code_ += "";
   }
 
+  // Split a SymbolTable of Value into key and value initializer strings. 
+  static void SerializeKeysAndValues(const SymbolTable<Value> &stable,
+                                     std::string *keys, std::string *vals) {
+    std::string sep = " ";
+    *keys = *vals = "{";
+    // TODO(joshv): the `dict` field does not preserve the insertion order
+    // of the symbols (their in-file order), and the `vec` field cannot be
+    // used to access both key and value in linear time. Add a `keys` field.
```

Comment:
We're not likely to add a keys field just for this use case. The order should not matter, in fact, I would say it be a good idea to ensure sorted order, so just use dict.